lib/rmt/downloader.rb

```diff
@@ -56,25 +61,30 @@ def process_queue
       local_file = make_local_path(remote_file)
     end while File.exist?(local_file) # rubocop:disable Lint/Loop
 
-    # The request is wrapped into a fiber for exception handling
-    request_fiber = Fiber.new do
-      begin
-        make_request(remote_file, local_file, request_fiber, queue_item[:checksum_type], queue_item[:checksum])
-      rescue RMT::Downloader::Exception => e
-        @logger.warn("× #{File.basename(local_file)} - #{e}")
-      ensure
-        process_queue
+    was_deduplicated = deduplicate(queue_item[:checksum_type], queue_item[:checksum], local_file)
```

Comment:
It probably makes sense to move that into a `# Skip over files that already exist` loop above.